File: Programación y Algoritmos/Actividad 18/imc.py
def CalcularIMC(peso: float, estatura: float) -> float:
    return peso/(estatura*estatura)

def EstadoDeSalud(imc: float) -> float:
    if imc < 18:
        print("Peso bajo. Necesario valorar signos de desnutrición")
    elif imc >= 18 and imc < 25:
        print("Peso normal")
    elif imc >= 25 and imc < 27:
        print("Sobrepeso")
    elif imc >= 27 and imc < 30: 
        print("Obesidad grado I. Riesgo relativo alto para desarrollar enfermedades cardiovasculares")
    elif imc >= 30 and imc < 40: 
        print("Obesidad grado II. Riesgo relativo muy alto para el desarrollo de enfermedades cardiovasculares")
    elif imc >= 40: 
        print("Obesidad grado III Extrema o Mórbida. Riesgo relativo extremadamente alto para el desarrollo de enfermedades cardiovasculares.")

    
peso = float(input("Introduzca el peso de la persona cuyo IMC quiere calcular en kg:  "))
estatura = float(input("Introduzca la estatura de la persona cuyo IMC desea calcular en metros:  "))
imc = CalcularIMC(peso, estatura)

print("El IMC de esta persona es de " + str(imc))
EstadoDeSalud(imc)

File: Programación y Algoritmos/Actividad 18/imc.cpp
#include <iostream>
#include <cmath>


using namespace std;

float CalcularIMC(float peso, float estatura);
void ImprimirEstadoDeSalud(float imc);

int main()
{
    float peso;
    float estatura;

    cout << "Introduzca el peso en kg y la estatura en metros de la persona cuyo IMC quiera calcular: " << endl;
    cin >> peso;
    cin >> estatura;
    cout << endl; 
    float imc = CalcularIMC(peso, estatura);

    cout << "El IMC de esta persona es " << imc << endl;
    cout << endl << "El peso de esta persona corresponde a: "; 

    ImprimirEstadoDeSalud(imc);
    cout << endl;
}


float CalcularIMC(float peso, float estatura)
{
    float imc = peso/(pow(estatura,2));

    return imc;
}

void ImprimirEstadoDeSalud(float imc)
{
    if (imc < 18)
    {
        cout << "Peso bajo" << endl; 
    }
    else if (imc >= 18 && imc < 25)
    {
        cout << "Peso normal" << endl; 
    }
    else if (imc >= 25 && imc < 27)
    {
        cout << "Sobrepeso" << endl;
    }
    else if (imc >= 27 && imc < 30)
    {
        cout << "Obesidad grado I" << endl;
    }
    else if (imc >= 30 && imc < 40)
    {
        cout << "Obesidad grado II" << endl;
    }
    else 
    {
        cout << "Obesidad grado III" << endl;
    }
}

